docs/architecture-decisions/0007-result-message-spec.md

```diff
@@ -0,0 +1,108 @@
+# 7. Result message spec
+
+Date: 2021-09-10
+
+## Status
+
+Accepted
+
+## Context
+
+Multiple external applications will read from the SQS result queues used by this
+service, so we need a specification to ensure consistent messages containing
+all data required for the consuming applications to process them.
+
+## Decision
+
+We will use the following result message specification:
+
+Each SQS Message sent by the dspace-submission-service to one of its result
+queues will contain two components, MessageAttributes and MessageBody.
+
+### MessageAttributes
+
+MessageAttributes is a JSON object containing two items, PackageID and
+SubmissionSource, structured like so:
+
+```json
+MessageAttributes = {
+    "PackageID": {
+        "DataType": "String",
+        "StringValue": "<A unique ID created by the submitting application that
+            will allow said application to match the result information to each
+            submitted package, e.g. 'etd_123123' or '98765'. The PackageID of
+            the result message must match the PackageID from the corresponding
+            submit message.>"
+    },
+    "SubmissionSource": {
+        "DataType": "String",
+        "StringValue": "<Name of the submitting system, e.g. 'ETD'. The
+          SubmissionSource of the result message must match the
+          SubmissionSource from the corresponding submit message.>"
+    }
+}
+```
+
+### MessageBody
+
+The MessageBody will contain detailed information about the submission result.
+The specification is slightly different for success results and error results.
+
+Note: SQS requires that the MessageBody be a string. However, this service and
+the submitting applications will want to parse/create the MessageBody as JSON
+objects, so the specification here shows the JSON object structure. This
+service must convert the MessageBody to a string before sending
+the message and the consuming applications must parse it back into a JSON
+object when processing messages from the result queue.
+
+#### Success Result
+
+```json
+MessageBody = {
+    "ResultType": "success",
```

Comment:
Did you consider putting the `ResultType` in the `MessageAttributes` section? It feels like it may be useful up there but I can't really pinpoint a reason it _needs_ to be different than this.

Reply:
I did. It ends up making slightly more sense this way because only the `MessageBody` changes based on the result success or failure, and the attributes are the same regardless.

Reply:
I see. So you are saying it's cleaner for DSS this way. I was thinking primarily of consuming applications. I'm good with this. Thanks for clarifying why you prefer it this way.